index.js

```diff
@@ -200,6 +200,11 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
     var ext
     var transform
     var inFile = path.resolve(inputDir, filename)
+
+    if (!_this.exclude.shouldInstrument(filename)) {
```

Comment:
refresh me if I'm wrong about how source-maps behave, but is it possible that a filename on disk would be excluded by our exclude rule, but has a filename the source-map maps to that isn't excluded?

we should look at the tests around excludeAfterInstrument behavior, which addresses some of these edge-cases, and make sure we continue to support these edge-cases.

Reply:
Maybe this should be a separate change?  I'm not sure how it's related to the main purpose of this PR 'enable existing options for stand-alone instrument command'.

Reply:
@coreyfarrell This change is pretty much the core of enabling include, exclude, excludeNegated functionality.  An assessment of whether the file should be instrumented is conducted in the function `_this.exclude.shouldInstrument(filename)`.  If it shouldn't be instrumented we return, if it should we continue and instrument the file.

@bcoe I've been looking over anything source-maps related and I'm still working on getting a good understanding of how everything hangs together.  So far I can see that the 'produce-source-maps' adds a source map while we're instrumenting, which shouldn't be an issue and seems unrelated to the 'source-map' option. 
I'm a little unsure of what 'source-map' should do in the instrument command.  My understanding is that it's mainly used to work with pre-instrumented files with existing source maps but I don't know why you'd be running the instrument command if you'd already instrumented the code elsewhere.  Anyway, my plan is to keep working on this until I'm confident it's good, any pointers, gotchas, or ideas for tests on this stuff would be appreciated.  Thanks